prover/src/lib.rs

```diff
@@ -79,6 +80,14 @@ pub type ReduceAir<F> = RecursionAir<F, REDUCE_DEGREE>;
 pub type CompressAir<F> = RecursionAir<F, COMPRESS_DEGREE>;
 pub type WrapAir<F> = RecursionAir<F, WRAP_DEGREE>;
 
+/// Contains result of execution along with an [ExecutionReport].
```

Comment:
nit: a bit weird to put this struct above `SP1Prover`

can you put it in `core`?